client/main.lua

```diff
@@ -0,0 +1,233 @@
+    end
+  end
+
+  if step == 'start' then
+    PersistentNotiMap[id] = true
+  elseif step == 'end' then
+    PersistentNotiMap[id] = false
+  end
+
+  if areTypesValid then
+    SendNUIMessage({
+      type = 'persistNoti',
+      step = step,
+      id = id,
+      options = options
+    })
+  end
+end
+--Initialize's Config after activated by Thread
+local function InitConfig()
+  local initObject = {
+    type = 'init',
+    position = cfg.position,
+    insertAnim = cfg.animations.insertAnimation,
+    insertDuration = cfg.animations.insertDuration,
+    removeAnim = cfg.animations.removeAnimation,
+    removeDuration = cfg.animations.removeDuration,
+    maxNotifications = cfg.maxNotifications
+  }
+  DebugPrint('Sending Init Config: \n' .. json.encode(initObject))
+  SendNUIMessage(initObject)
+end
+
+RegisterNUICallback('nuiReady', function(_, cb)
+  DebugPrint('NUI frame ready')
+  nuiReady = true
+  -- Send Config File after NUI frame ready
+  InitConfig()
+  cb({})
+end)
+
+--OBJECT STYLED EXPORTS
+function Alert(data)
+  SendNotification({
+    type = 'noti',
+    style = data.style,
+    duration = data.duration,
+    title = nil,
+    message = data.message,
+    image = nil,
+    sound = data.sound,
+    custom = data.custom,
+    position = data.position,
+    icon = data.icon
+  })
+end
+
+exports('Alert', Alert)
+
+function Custom(data)
+  SendNotification({
+    type = 'noti',
+    style = data.style,
+    duration = data.duration,
+    title = data.title,
+    message = data.message,
+    image = data.image,
+    sound = data.sound,
+    custom = data.custom,
+    position = data.position,
+    icon = data.icon
+  })
+end
+
+exports('Custom', Custom)
+
+function Image(data)
+  SendNotification({
+    type = 'noti',
+    style = data.style,
+    duration = data.duration,
+    title = data.title,
+    message = nil,
+    image = data.image,
+    sound = data.sound,
+    custom = data.custom,
+    position = data.position,
+    icon = nil
+  })
+end
+
+exports('Image', Image)
+
+function Icon(data)
+  SendNotification({
+    type = 'noti',
+    style = data.style,
+    duration = data.duration,
+    title = data.title,
+    message = data.message,
+    image = nil,
+    sound = data.sound,
+    custom = data.custom,
+    position = data.position,
+    icon = data.icon
+  })
+end
+
+exports('Icon', Icon)
+
+function Persist(data)
+  SendPersistentNotification(data.step, data.id, data.options)
+end
+
+exports('Persist', Persist)
+
+function IsPersistentShowing(id)
+  return PersistentNotiMap[id] or false
+end
+
+exports('IsPersistentShowing', IsPersistentShowing)
+
+--Event Handlers from Server (Objects)
+RegisterNetEvent('t-notify:client:Alert', function(data)
+  Alert(data)
+end)
+
+RegisterNetEvent('t-notify:client:Custom', function(data)
+  Custom(data)
+end)
+
+RegisterNetEvent('t-notify:client:Image', function(data)
+  Image(data)
+end)
+
+RegisterNetEvent('t-notify:client:Icon', function(data)
+  Icon(data)
+end)
```

Comment:
I didn't realize there wasn't any manipulation in these blocks on my last review 😅

```suggestion
RegisterNetEvent('t-notify:client:Alert', Alert)
RegisterNetEvent('t-notify:client:Custom', Custom)
RegisterNetEvent('t-notify:client:Image', Image)
RegisterNetEvent('t-notify:client:Icon', Icon)
```

Reply:
Yep. I think I also mentioned the manipulation was done on the function itself.